core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -74,17 +74,28 @@ object StaticFile {
       }
     })
 
-  def fromFile[F[_]: Sync](f: File, req: Option[Request[F]] = None): OptionT[F, Response[F]] =
-    fromFile(f, DefaultBufferSize, req)
+  def calcETag: File => String = _.length().toString
```

Comment:
`File.length()` can throw and should probably be wrapped in `F`.

Reply:
It's only ever called from inside F.delay

Reply:
Not referentially transparent is not referentially transparent.

Reply:
I didn't get you, all the file calls aren't, they are all inside one F.delay block.

Reply:
This function is not referentially transparent. It is a top level public function that does not respect its own function signature. If it was used safely everywhere or private that would not change the fact that the function itself does not respect its own signature.